src/julia-syntax.scm

```diff
@@ -1602,13 +1602,14 @@
                              ,body))
                          `(scope-block ,body))))
                `(block (= ,coll ,(car itrs))
-                       (= ,state (call (top start) ,coll))
+                       (= ,next (call (top iterate) ,coll))
                        ;; TODO avoid `local declared twice` error from this
                        ;;,@(if outer `((local ,lhs)) '())
                        ,@(if outer `((require-existing-local ,lhs)) '())
                        (_while
-                        (call (top not_int) (call (core typeassert) (call (top done) ,coll ,state) (core Bool)))
-                        ,body))))))))
+                        (call (top not_int) (call (core ===) ,next (null)))
```

Comment:
Interesting – why is `not_int` getting imported into `top`? This should be found in `(|.| (core Intrinsics) 'not_int)`, and not found in `top`.

Reply:
Base probably imports it.